sessionctx/stmtctx/stmtctx.go

```diff
@@ -104,6 +104,8 @@ func (warn *SQLWarn) UnmarshalJSON(data []byte) error {
 // StatementContext contains variables for a statement.
 // It should be reset before executing a statement.
 type StatementContext struct {
+	// Expired will be true when a new query comes but the current StmtCtx is not reset yet
+	Expired bool
```

Comment:
Not easy to do this because after the first statement "reset" a stmt context and the stmt finished, we should change the `NotReset` to false before `RestStatementContext` of the next statement. It is the same of the current implement

Reply:
I still doubts this will race somehow. Maybe it is enough to copy some of the property instead of directly `&sc`. Primitives like string or integer have little performance impact.